### PR TITLE
modules: openthread: platform: radio: Time Sync OT Feature support

### PR DESCRIPTION
**Pull request by ghost**
Add missing otRadioIeInfo structure and add plaform time to transmit frame

### CODE REVIEW

File: modules/openthread/platform/radio.c
Comment (rlubos) [inline modules/openthread/platform/radio.c:405]: Doesn't this duplicate the write at line 398 (I like this approach better though)?

File: modules/openthread/platform/radio.c
Comment (rlubos) [inline modules/openthread/platform/radio.c:411]: Are we just writing 64-bit integer to the buffer in little endian here? We have `sys_put_le64()` helper in Zephyr (https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/sys/byteorder.h#L485).

Comment (DaveLacerteHQ) [reply modules/openthread/platform/radio.c]: Changed for sys_put_le64. Thanks for the hint. Way cleaner!

File: modules/openthread/platform/radio.c
Comment (rlubos) [inline modules/openthread/platform/radio.c:398]: @DaveLacerteHQ In my [previous comment](https://github.com/zephyrproject-rtos/zephyr/pull/70758#discussion_r1540664551) I was referring to this line, not the line below where the pointer is initialized.

Comment (DaveLacerteHQ) [reply modules/openthread/platform/radio.c]: You are right. Sorry for the confusion. Corrected.

**Comment by canisLupus1313** [comment]
@DaveLacerteHQ  Time sync is experimental feature of Openthread and not a part of Thread specification. Would be nice if you could add `EXPERIMANTAL` to `OPENTHREAD_TIME_SYNC`. 
https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/openthread/Kconfig.features#L343

**Review by rlubos — DISMISSED** [review]
Techically looks good, but would be good to get second review from someone more familiar with those areas of OT stack.

**Comment by DaveLacerteHQ** [comment]
> @DaveLacerteHQ Time sync is experimental feature of Openthread and not a part of Thread specification. Would be nice if you could add `EXPERIMANTAL` to `OPENTHREAD_TIME_SYNC`. https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/openthread/Kconfig.features#L343

Done

**Review by canisLupus1313 — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**
